### PR TITLE
refactor hvdc logs for creation and modification

### PR DESCRIPTION
**Pull request by EtienneLt**
refactor hvdc logs for properties for creation and modification

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/modifications/PropertiesUtils.java [hunk 1]
```diff
@@ -21,32 +20,71 @@ public static void applyProperties(Identifiable<?> identifiable, ReportNode subR
...
     }
+
+    public static void reportPropertiesInfos(Identifiable<?> identifiable, List<FreePropertyInfos> modificationInfosProperties,
+                                             String propertiesLabelKey, ReportNode subReportNode) {
+        Optional<List<FreePropertyInfos>> properties = Optional.ofNullable(modificationInfosProperties);
+        List<ReportNode> reportNodes = new ArrayList<>();
+        if (properties.isPresent()) {
+            ReportNode propertiesReporter = subReportNode.newReportNode().withMessageTemplate(propertiesLabelKey, PROPERTIES).add();
+            properties.get().forEach(property -> reportNodes.add(applyPropertyReportNode(identifiable, property)));
+            ModificationUtils.getInstance().reportModifications(propertiesReporter, reportNodes,
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/PropertiesUtils.java:48]: It's weird to have another sub report node with message="Properties" here. But we have this problem for every sub section. We should fix it everywhere in another PR

File: src/main/java/org/gridsuite/modification/server/modifications/PropertiesUtils.java [hunk 2]
```diff
@@ -21,32 +20,71 @@ public static void applyProperties(Identifiable<?> identifiable, ReportNode subR
...
+    }
+
+    private static void modifyPropertyReportNode(ReportNodeAdderOrBuilder<?> adderOrBuilder, FreePropertyInfos prop) {
+        adderOrBuilder.withMessageTemplate("propertyAdded", "    Property ${name} added with value ${value}")
+            .withUntypedValue("name", prop.getName())
+            .withUntypedValue("value", prop.getValue())
+            .withSeverity(TypedValue.INFO_SEVERITY);
+    }
+
+    private static void deletePropertyReportNode(ReportNodeAdderOrBuilder<?> adderOrBuilder, FreePropertyInfos prop) {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/PropertiesUtils.java:85]:
```suggestion
    private static void reportPropertyDeletion(ReportNodeAdderOrBuilder<?> adderOrBuilder, FreePropertyInfos prop) {
```

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/modifications/PropertiesUtils.java (hunk 1 above)
Comment (EtienneLt) [reply src/main/java/org/gridsuite/modification/server/modifications/PropertiesUtils.java:48]: yes I agree

File: src/main/java/org/gridsuite/modification/server/modifications/PropertiesUtils.java [hunk 3]
```diff
@@ -1,52 +1,70 @@
...
+    public static final String PROPERTIES = "Properties";
 
     private PropertiesUtils() {
         // Should not be instantiated
     }
 
-    public static void applyProperties(Identifiable<?> identifiable, ReportNode subReportNode, @Nullable List<FreePropertyInfos> properties) {
+    public static void applyProperties(Identifiable<?> identifiable, ReportNode subReportNode, @Nullable List<FreePropertyInfos> properties, String propertiesLabelKey) {
+        List<ReportNode> reportNodes = new ArrayList<>();
+        ReportNode propertiesReporter = subReportNode.newReportNode().withMessageTemplate(propertiesLabelKey, PROPERTIES).add();
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/modifications/PropertiesUtils.java:19]:
```suggestion
        ReportNode propertiesReporter = subReportNode.newReportNode().withMessageTemplate(PROPERTIES, PROPERTIES).add();
```
So you do the same as it is done for setpoints or limits. And you need a new parameter in applyProperties()

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**